jdaviz/utils.py

```diff
@@ -385,3 +390,52 @@ def total_masked_first_data(self):
     def __setgluestate__(cls, rec, context):
         masks = {key: context.object(value) for key, value in rec['masks'].items()}
         return cls(masks=masks)
+
+
+def download_uri_to_path(possible_uri, cache=False, local_path=None):
```

Comment:
Maybe default should be to always use cache, if available?
```suggestion
def download_uri_to_path(possible_uri, cache=True, local_path=None):
```

Reply:
I prefer to keep it false by default – for users who cache a lot, it can be surprising and really frustrating to have significant disk space used by files without human readable names, making it hard to clean up without deleting all.

Reply:
But the other side is that you keep downloading the same thing over and over again, which can get expensive via AWS.

Reply:
If our files were guaranteed to be small, I wouldn't blink. But I have cached a bunch of 5 GB NIRCam images, and regret that I did.

Reply:
I also have regretted accidentally re-downloading GB of data I already have. So, I don't think there is a perfect solution.

Now I think we should have a user doc about this new feature and document all the gotchas there.

Reply:
I don't have strong feelings either way, but here's a third option: what if, since there are good arguments both ways, we require the user to explicitly set the `cache` argument in the case of loading from a URL? We could default to `cache=None` rather than `True` or `False` and then check to see if it's `None` in the appropriate case and raise an error telling the user to set it explicitly.

Reply:
@rosteen , but you still need a default for "standalone app", do you not?

Reply:
Hmm yes that's true 😭